app/src/main/java/org/wikipedia/recurring/CategoriesTableCleanupTask.kt

```diff
@@ -1,35 +1,28 @@
 package org.wikipedia.recurring
 
-import org.wikipedia.WikipediaApp
 import org.wikipedia.database.AppDatabase
 import org.wikipedia.util.log.L
-import java.util.Calendar
+import java.time.LocalDateTime
 import java.util.Date
 import java.util.concurrent.TimeUnit
 
-class CategoriesTableCleanupTask(app: WikipediaApp) : RecurringTask() {
+class CategoriesTableCleanupTask : RecurringTask() {
     override val name = "categoriesTableCleanupTask"
 
     override fun shouldRun(lastRun: Date): Boolean {
         return millisSinceLastRun(lastRun) >= TimeUnit.DAYS.toMillis(DAYS_BETWEEN_RUNS)
     }
 
     override suspend fun run(lastRun: Date) {
-        val twoYearsAgoTimeStamp = getTimeStampForYearsAgo()
+        val twoYearsAgoTimeStamp = LocalDateTime.now().year - CLEANUP_TIME_IN_YEARS
         deleteOldDataInBatches(twoYearsAgoTimeStamp, 1000)
         L.d("Successfully deleted Category data older than $CLEANUP_TIME_IN_YEARS years")
     }
 
-    private fun getTimeStampForYearsAgo(years: Int = CLEANUP_TIME_IN_YEARS): Long {
-        val calendar = Calendar.getInstance()
-        calendar.add(Calendar.YEAR, -years)
-        return calendar.timeInMillis
-    }
-
-    suspend fun deleteOldDataInBatches(timeStamp: Long, batchSize: Int) {
+    suspend fun deleteOldDataInBatches(year: Int, batchSize: Int) {
         var deletedCount = 0
         do {
-            deletedCount = AppDatabase.instance.categoryDao().deleteOlderThanInBatch(timeStamp, batchSize)
+            deletedCount = AppDatabase.instance.categoryDao().deleteOlderThanInBatch(year, batchSize)
```

Comment:
I don't really think we need the "batch" logic any longer, nor to have a `do { }` loop here. Sqlite is very efficient, and should not have any trouble deleting thousands of entries in one call.